File: Python/remove_repeating_characters.py
#!/usr/bin/env python

def remove_repeats(string: str) -> str:
    chars = []
    for c in string:
        if len(chars) == 0 or c != chars[-1]:
            chars.append(c)
    return "".join(chars)

assert remove_repeats('') == ''
assert remove_repeats('a') == 'a'
assert remove_repeats('abbcccda') == 'abcda'

print('SUCCESS')

File: Haskell/remove_repeating_characters.hs
import qualified Control.Exception
assert = flip Control.Exception.assert (pure ())

--------------------------------------------------------------------------------

removeRepeats :: String -> String
removeRepeats "" = ""
removeRepeats [x] = [x]
removeRepeats (x0:x1:xs) = 
    if x0 == x1
    then removeRepeats (x0:xs)
    else x0 : removeRepeats (x1:xs)


main = do
    assert $ removeRepeats "" == ""
    assert $ removeRepeats "a" == "a"
    assert $ removeRepeats "abbcccda" == "abcda"

    putStrLn "SUCCESS"
    
 
